utils/crash/logger.go

```diff
@@ -0,0 +1,56 @@
+package crash
+
+import (
+	"net/http"
+	"runtime"
+	"runtime/debug"
+	"sync"
+
+	"github.com/rudderlabs/rudder-go-kit/logger"
+)
+
+type panicLogger struct {
+	notifyOnce sync.Once
+	logger     logger.Logger
+
+	opts PanicWrapperOpts
+}
+
+// UsingLogger uses the provided logger to log panics.
+func UsingLogger(logger logger.Logger, opts PanicWrapperOpts) *panicLogger {
+	return &panicLogger{
+		logger: logger.Child("panic"),
+		opts:   opts,
+	}
+}
+
+// Notify returns a function that should be deferred to capture panics.
+// It will do a Fatal log with the stack trace and panic value.
+func (p *panicLogger) Notify(team string) func() {
+	return func() {
+		if r := recover(); r != nil {
+			p.notifyOnce.Do(func() {
+				p.logger.Fatalw("Panic detected. Application will crash.",
+					"stack", string(debug.Stack()),
+					"panic", r,
+					"team", team,
+					"goRoutines", runtime.NumGoroutine(),
+					"version", p.opts.AppVersion,
+					"releaseStage", p.opts.ReleaseStage,
+					"appType", p.opts.AppType,
+				)
```

Comment:
We can use a non-sugared structured log in here.
```suggestion
				p.logger.Fataln("Panic detected. Application will crash.",
					logger.NewStringField("stack", string(debug.Stack())),
					logger.NewField("panic", r),
					logger.NewStringField("team", team),
					logger.NewIntField("goRoutines", int64(runtime.NumGoroutine())),
					logger.NewStringField("version", p.opts.AppVersion),
					logger.NewStringField("releaseStage", p.opts.ReleaseStage),
					logger.NewStringField("appType", p.opts.AppType),
				)
```